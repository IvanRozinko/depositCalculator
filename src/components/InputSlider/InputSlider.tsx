import React, { ReactNode } from 'react';
import { Slider, Box, Divider } from '@material-ui/core';
import { Theme, createStyles, makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) => 
  createStyles({
    container: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      [theme.breakpoints.down('xs')]: {
        flexDirection: 'column',
        alignItems: 'stretch',
        padding: '20px 0'
      }
    },
    slider: {
      display: 'flex',
      flexGrow: 1,
      margin: '0 20px',
      padding: '20px 0',
      [theme.breakpoints.down('xs')]: {
        margin: 0,
        paddingTop: 0
      }
    }
  })
)

interface Props {
  value: number,
  handleChange: (e: any, value: number | number[]) => void,
  children?: ReactNode,
  marks?: { value: number, label: string }[],
  rate?: number,
  label: ReactNode,
  valueDisplay: 'on' | 'off',
  sliderOptions: { min: number, max: number, 'aria-labelledby': string }
}

function InputSlider({ 
  value, handleChange, marks, rate, label, sliderOptions, valueDisplay, children=null 
}: Props) {

  const styles = useStyles();
  
  const valueLabelFormat = () => rate;

  return (
    <>
      <Box
        className={styles.container}
      >
        {label}
          <div className={styles.slider}>
            <Slider 
              value={value} 
              onChange={handleChange}
              aria-labelledby={sliderOptions['aria-labelledby']}
              min={sliderOptions.min}
              max={sliderOptions.max}
              marks={marks}
              valueLabelDisplay={valueDisplay}
              valueLabelFormat={valueLabelFormat}
            />
          </div>
          {children}
      </Box>
      <Box display={{ xs: 'block', sm: 'none' }}>
        <Divider />
      </Box>
    </>
  )
}

export default InputSlider
