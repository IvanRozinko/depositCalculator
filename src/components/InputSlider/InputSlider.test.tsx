import InputSlider from "./InputSlider";
import { shallow } from 'enzyme';
import { Slider } from '@material-ui/core';

describe('<InputSlider />:', () => {
  const value = 10;
  const handleChange = jest.fn();
  const label = null;
  const valueDisplay = 'off';
  const sliderOptions = {
    min: 0,
    max: 10,
    'aria-labelledby': 'stringg' 
  }
  const wrapper = shallow(
    <InputSlider 
      value={value}
      handleChange={handleChange}
      label={label}
      valueDisplay={valueDisplay}
      sliderOptions={sliderOptions}
    />
  )

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleChange onChange', () => {
    const event = new Event('onchange')
    const newValue = 12;
    wrapper.find(Slider).simulate('change', event, newValue)
    expect(handleChange).toBeCalledWith(event, newValue);
  }) 
})