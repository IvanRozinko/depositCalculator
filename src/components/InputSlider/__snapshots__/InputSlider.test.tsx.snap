// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSlider />: should renders correctly 1`] = `
<Fragment>
  <Styled(MuiBox)
    className="makeStyles-container-1"
  >
    <div
      className="makeStyles-slider-2"
    >
      <WithStyles(ForwardRef(Slider))
        aria-labelledby="stringg"
        max={10}
        min={0}
        onChange={[MockFunction]}
        value={10}
        valueLabelDisplay="off"
        valueLabelFormat={[Function]}
      />
    </div>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    display={
      Object {
        "sm": "none",
        "xs": "block",
      }
    }
  >
    <WithStyles(ForwardRef(Divider)) />
  </Styled(MuiBox)>
</Fragment>
`;
